File: output/CodeGen/avatar/Java/Python/atcoder_ABC128_C.py
def main():
    sc = Scanner()
    N = sc.nextInt()
    M = sc.nextInt()
    a = [0] * N
    for i in range(N):
        a[i] = 0
    for i in range(M):
        k = sc.nextInt()
        for j in range(k):
            s = sc.nextInt()
            s -= 1
            a[s] |= 1 << i
    p = 0
    for i in range(M):
        x = sc.nextInt()
        p |= (x << i)
    ans = 0
    for s in range(2**N):
        t = 0
        for i in range(N):
            if ( ( ( s >> i ) & 1 ) == 1 ):
                t ^= a[i]
        if ( p == t ):
            ans += 1
    print(ans)
    sc.close()

File: output/StarCoder/avatar/Python/Java/atcoder_ABC128_C.java
import java.util.*;
public class atcoder_ABC128_C {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();
        int[][] li = new int[m][n];
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                li[i][j] = sc.nextInt();
            }
        }
        int[] p = new int[n];
        for (int i = 0; i < n; i++) {
            p[i] = sc.nextInt();
        }
        Set<Integer> q = new HashSet<>();
        Set<Integer> r = new HashSet<>();
        for (int i = 0; i < m; i++) {
            for (int j = 1; j < n; j++) {
                q.add(li[i][j]);
            }
        }
        for (int i = 1; i <= n; i++) {
            r.add(i);
        }
        r.removeAll(q);
        int ans = 0;
        for (int i = 0; i <= q.size(); i++) {
            for (Set<Integer> s : Sets.powerSet(q)) {
                if (s.size() % 2!= p[i]) {
                    break;
                }
            } else {
                ans += Math.pow(2, r.size());
            }
        }
        System.out.println(ans);
    }
}